From 0d21c0562f778e75c120a131d7c59b0220e27548 Mon Sep 17 00:00:00 2001
From: Przemyslaw Czarnota <przemyslawx.czarnota@intel.com>
Date: Wed, 13 May 2020 09:13:07 +0200
Subject: [PATCH] PONRTSYS-7005: Enable PON Qos driver

- Remove fallback to QoS Manager

The fallback was added, because PON QoS didn't support all features, add
we needed dp_ndo_setup_tc() to fall back to QoS Manager in this case.
Now PON QoS supports all neccessary features, so the fallback can be
removed.
---
 drivers/net/datapath/dpm/datapath_instance.c | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/drivers/net/datapath/dpm/datapath_instance.c b/drivers/net/datapath/dpm/datapath_instance.c
index 0628567cfaab..a99a3ad779b4 100644
--- a/drivers/net/datapath/dpm/datapath_instance.c
+++ b/drivers/net/datapath/dpm/datapath_instance.c
@@ -187,22 +187,21 @@ int (*pon_qos_setup_tc_fn)(struct net_device *dev,
 			   int port_id,
 			   int deq_idx);
 EXPORT_SYMBOL(pon_qos_setup_tc_fn);
+#define DP_ENABLE_TC_OFFLOADS
+#endif
 
+#if (IS_ENABLED(CONFIG_QOS_MGR) && !IS_ENABLED(CONFIG_GRX500_CBM))
+#define DP_ENABLE_TC_OFFLOADS
 #endif
 
-#if IS_ENABLED(CONFIG_QOS_MGR)
+#if defined(DP_ENABLE_TC_OFFLOADS)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0)
 static int dp_ndo_setup_tc(struct net_device *dev, u32 handle,
 			   __be16 protocol, struct tc_to_netdev *tc)
 {
 #if IS_ENABLED(CONFIG_PON_QOS)
-	int ret = -1;
-
-	if (pon_qos_setup_tc_fn) {
-		ret = pon_qos_setup_tc_fn(dev, handle, protocol, tc, -1, -1);
-		if (!ret)
-			return ret;
-	}
+	if (pon_qos_setup_tc_fn)
+		return pon_qos_setup_tc_fn(dev, handle, protocol, tc, -1, -1);
 #endif
 
 #if IS_ENABLED(CONFIG_QOS_MGR)
@@ -236,7 +235,7 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 	struct dp_dev *dp_dev;
 	int new_f = 0;
 	u32 idx;
-#if IS_ENABLED(CONFIG_QOS_MGR) && !IS_ENABLED(CONFIG_GRX500_CBM)
+#if defined(DP_ENABLE_TC_OFFLOADS)
 	int offset;
 #endif
 
@@ -272,7 +271,7 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 			hlist_add_head(&dp_dev->hlist, &dp_dev_list[idx]);
 		}
 
-#if IS_ENABLED(CONFIG_QOS_MGR) && !IS_ENABLED(CONFIG_GRX500_CBM)
+#if defined(DP_ENABLE_TC_OFFLOADS)
 		dev->features |= NETIF_F_HW_TC;
 		offset = offsetof(const struct net_device_ops, ndo_setup_tc);
 		dp_set_net_dev_ops_priv(dev, &dp_ndo_setup_tc, offset,
@@ -299,7 +298,7 @@ int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
 		    int ep, u16 ctp, u32 flag)
 {
 	struct dp_dev *dp_dev;
-#if IS_ENABLED(CONFIG_QOS_MGR) && !IS_ENABLED(CONFIG_GRX500_CBM)
+#if defined(DP_ENABLE_TC_OFFLOADS)
 	int offset;
 #endif
 
@@ -358,7 +357,7 @@ int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
 #endif
 			dp_dev->owner &= ~DP_OPS_OWNER_DPM;
 
-#if IS_ENABLED(CONFIG_QOS_MGR) && !IS_ENABLED(CONFIG_GRX500_CBM)
+#if defined(DP_ENABLE_TC_OFFLOADS)
 			offset = offsetof(const struct net_device_ops,
 					  ndo_setup_tc);
 			dp_set_net_dev_ops_priv(dev, NULL,
